tests/spec_decode/e2e/test_multistep_correctness.py

```diff
@@ -11,9 +11,15 @@
 numerics, see https://arxiv.org/pdf/2302.01318.pdf), we can expect greedy
 equality. This gives us good coverage of temp=0.
 
+At temp=0, the TypicalAcceptanceSampler ensures that only the tokens with the
+highest probability in the target distribution are accepted. Therefore, we can 
+expect greedy equality for the TypicalAcceptanceSampler at temp=0.
+
 For temp>0, we rely on unit tests on the rejection sampler to verify that the
 output distribution is the same with spec decode vs. no spec decode (this would
-be prohibitively expensive to run with a real model).
+be prohibitively expensive to run with a real model). Similary, for the
+TypicalAcceptance sampler also, we rely on unit tests to validate temp>0
+test cases.
```

Comment:
For testing strategy:

I am concerned that we are adding many E2E tests that don't provide a lot of signal over what already exists. The tradeoff of more tests is that we can accidentally explode CI time. This is because we rely on E2E tests for spec decode correctness and any small regression in model loading or vLLM initialization time can hurt us bad.

So, what I suggest:
* E2E tests over the interaction between spec decode worker and typical acceptance
  * make sure it can handle different BS
  * make sure it can handle different K
* E2E test over one other proposer method
  * just need one to make sure typical acceptance works beyond draft model
* We _don't_ need tests around preemption, disabling/skipping speculation, different block size, since all of these are no different for typical acceptance.

Reply:
Added a single test in test_multistep_correctness.py to cover different batch size and speculation_length values with TypicalAcceptanceSampler. Added a similar test to test_ngram_correctness.py